libvast/src/system/disk_monitor.cpp

```diff
@@ -110,11 +108,15 @@ disk_monitor(disk_monitor_actor::stateful_pointer<disk_monitor_state> self,
           VAST_VERBOSE("{} failed to find partition {}", self, partition);
           continue;
         }
-        // TODO: Wrap a more generic `stat()` using `vast::path`.
-        struct stat statbuf;
-        if (::stat(file.path().c_str(), &statbuf) < 0)
-          continue;
-        partitions.push_back({id, statbuf.st_size, statbuf.st_mtime});
+        std::error_code ec{};
+        const auto file_size = std::filesystem::file_size(file.path(), ec);
+        const auto mtime = std::filesystem::last_write_time(file.path(), ec);
```

Comment:
Didn't you want to switch this to `file.file_size(ec)`/`file.last_write_time(ec)` after #1411 has been merged?

Reply:
Yes, thanks for the reminder. I will make that change later tonight -- sorry about that.

Reply:
CI is passing now with using the `file_size` and `last_write_time` directly as member function calls from the corresponding `directory_entry` type as discussed.

Reply:
ping @lava -- how does this look to you?